grid/grid_network.py

```diff
@@ -55,19 +114,98 @@ def serve_model(self, model, model_id):
             host_worker.serve_model(model, model_id=model_id)
             host_worker.disconnect()
 
-    def run_inference(self, model_id, dataset):
+    def run_inference(self, model_id, dataset, encrypted=False, copy=True):
```

Comment:
can you rename dataset to data? Let's try to keep this API as close to the grid nodes API as possible